vllm/sequence.py

```diff
@@ -1112,6 +1139,33 @@ class ExecuteModelRequest:
     num_steps: int = 1
     # Finished request ids since last step.
     finished_requests_ids: List[str] = field(default_factory=list)
+    # The last sampled token ids for multi step decoding.
+    last_sampled_token_ids: Optional[torch.Tensor] = None
+
+    @property
+    def is_first_multi_step(self) -> bool:
+        # TODO(will) make this be able to handle batches with variable number of
+        # steps
+        assert len(self.seq_group_metadata_list) > 0
+        first_seq_group = self.seq_group_metadata_list[0]
+        return first_seq_group.state.current_step == 0
```

Comment:
so this means that the state of multi step is decided by the state of the first sequence group in a batch? E.g., if the first seq group has only 1 step left, we only run 1 step although num_steps > 1?

Reply:
yes, eventually we may change this behavior and allow sequences in a batch to be different steps

Reply:
Is this really a necessary feature?